### PR TITLE
update Cremona automatically if necessary 

### PR DESCRIPTION
**Pull request by giovannistagliano**

### CODE REVIEW

**Comment by mahrud** [comment]
Just a suggestion: this would be a cool feature to have in the Core, but I think a lot of things can go wrong with this implementation.

**Comment by giovannistagliano** [comment]
> Just a suggestion: this would be a cool feature to have in the Core, but I think a lot of things can go wrong with this implementation.

Thanks for your comment. Yes, it should be nice to have a function of this kind in the core. 
However, my implementation is very experimental. I don't even know if it works on operating systems different from Linux.
It would also be nice that it automatically check the last version on github.

**Comment by DanGrayson** [comment]
Now what?

```
[9/553] Building CXX object Macaulay2/system/CMakeFiles/M2-supervisor.dir/supervisor.cpp.o
FAILED: Macaulay2/system/CMakeFiles/M2-supervisor.dir/supervisor.cpp.o 
/usr/local/bin/ccache /usr/local/opt/ccache/libexec/clang++  -I../../Macaulay2/system -I../../Macaulay2/c -I../../Macaulay2/e -Os -DNDEBUG -isysroot /Applications/Xcode_12.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -mmacosx-version-min=10.15 -DNDEBUG -DOM_NDEBUG -DSING_NDEBUG -Wuninitialized -g --sysroot=/Applications/Xcode_12.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -I/Users/runner/work/M2/M2/M2/BUILD/build/usr-host/include -I/Users/runner/work/M2/M2/M2/include -I/Users/runner/work/M2/M2/M2/BUILD/build/include -Wall -Wextra -Wfatal-errors -Wcast-qual -Wno-unused-parameter -Wno-attributes -Wno-unused -std=gnu++14 -MD -MT Macaulay2/system/CMakeFiles/M2-supervisor.dir/supervisor.cpp.o -MF Macaulay2/system/CMakeFiles/M2-supervisor.dir/supervisor.cpp.o.d -o Macaulay2/system/CMakeFiles/M2-supervisor.dir/supervisor.cpp.o -c ../../Macaulay2/system/supervisor.cpp
In file included from ../../Macaulay2/system/supervisor.cpp:1:
../../Macaulay2/system/supervisor.hpp:5:10: fatal error: 'atomic_ops.h' file not found
#include <atomic_ops.h>
         ^~~~~~~~~~~~~~
1 error generated.
[10/553] Building CXX object Macaulay2/system/CMakeFiles/M2-supervisor.dir/m2file.cpp.o
[11/553] Building C object Macaulay2/c/CMakeFiles/scc1.dir/cprint.c.o
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.
```

**Comment by giovannistagliano** [comment]
Now the function updatePackage seems work quite well. I think it would be useful to further refine it and include it in the Core.

File: M2/Macaulay2/packages/MultiprojectiveVarieties.m2 [hunk 1]
```diff
@@ -22,11 +22,101 @@ newPackage(
...
+        for f in listauxfiles do removeFile(dir|packagename|"/"|f|".m2"); 
+        removeDirectory(dir|packagename);  
+    ); 
+    <<"--** package "<<packagename<<", version "<<v0<<", has been successfully installed **--"<<endl;
+    get("!"|///M2 -e "loadPackage \"///|packagename|///\"; \"///|dir|///packageVersion.m2\"<<///|packagename|///.Options.Version<<close;" 2>&1 &///);
+    v := get(dir|"packageVersion.m2");
+    removeFile(dir|"packageVersion.m2");
+    if v != v0 then (
+        sf := (value packagename)#"source file";
+        err := "Error in loading the new version of the package "|packagename|". Check that in your PATH you do not have any older version of the package.";
```
Comment (DanGrayson) [inline M2/Macaulay2/packages/MultiprojectiveVarieties.m2:63]: Change PATH to 'path', as PATH refers to an environment variable, and path refers to a Macaulay2 variable.

**Pull request merged**